### DISCUSSION TITLE
Did anybody set up eslint flat config with eslint lsp extra?

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2732] [Q&A] [answered]
Hi everyone! I used eslint LSP for long time. As for now, it doesn't respect the new eslint flat config format. I tried to debug it within vscode because as far as I know, mason eslint lsp use the vscode eslint lsp. And yeah, it can't find eslint.config.js while .eslintrc.js finds just fine

https://github.com/LazyVim/LazyVim/assets/51330172/e6a0c7dd-1b7e-44a3-9d9d-f60b7b005214

I managed to get it to work with vscode by adding `"eslint.experimental.useFlatConfig": true` to vscode settings json file. I added the same thing to the eslint initialization options:
```lua
return {
  'neovim/nvim-lspconfig',
  -- other settings removed for brevity
  opts = {
    ---@type lspconfig.options
    servers = {
      eslint = {
        settings = {
          -- helps eslint find the eslintrc when it's placed in a subfolder instead of the cwd root
          workingDirectories = { mode = 'auto' },
          experimental = {
            -- allows to use flat config format
            useFlatConfig = true,
          },
        }
      },
    },
  }
}
```

But it doesn't work for whatever reason. Has anyone configured the eslint flat config to work with eslint LSP or with eslint_d?

P.S. Does anyone know any plugin that helps to see debug logs per each LSP?

**Comment by madflow** [comment]
I did a quick test, and it just works. I use the "extras.linting.eslint" rule. Maybe you could be more speficic what is "not working"?

**Reply by Demianeen** [reply]
Hi yeah, it does not give any listing errors, or any errors, even in lsp log, except for the warning about missing action workspace/didChangeWorkspaceFolders. But it is always there because it is not VSCode.
![CleanShot 2024-03-16 at 10 29 19@2x](https://github.com/LazyVim/LazyVim/assets/51330172/18dce675-4d89-4703-88b5-601cb3da805d)


https://github.com/LazyVim/LazyVim/assets/51330172/8b01e414-a3a7-4b8e-b883-8a7de9b2043d

Also, could you let me know if you tried with flat config?

It should look like this:
```
module.exports = [
...
]
```

Sorry to ask, but I had a few other people who assumed I was talking about the older eslint config format.

**Reply by madflow** [reply]
Well I put 

```
export default [
    {
        files: ["**/*.js"],
        rules: {
            "semi": "error",
            "no-unused-vars": "error"
        }  
    }
];
```

in a eslint.config.js file and it worked. I just did what was desribed here: https://eslint.org/blog/2022/08/new-config-system-part-2/

In the video you posted is "hard" parse error by eslint. This could be the problem, when eslint does an hard exit. Maybe move the file in question (tsconfig.base.json) and try again.

**Reply by Demianeen** [reply, marked as answer]
Thanks @madflow! Thanks to you I figured out that the issue was the eslint config itself. It is strange because it worked fine with the same configuration with .eslintrc 😅

It is also super strange that there were no error in lsp debug log about error with starting eslint

**Comment by sondrej** [comment]
Related issue with latest version of vscode-langservers-extracted: https://github.com/microsoft/vscode-eslint/issues/1839
After updating this through Mason from 4.8.0 to 4.10.0 I no longer get diagnostics/errors from eslint. Apparently `workingDirectories = { mode = 'auto' },` seems to be broken in latest release.

**Reply by sondrej** [reply]
Only workaround I've found so far is to downgrade eslint-lsp to version 4.8.0:
`: MasonInstall eslint-lsp@4.8.0`

**Reply by qadzifi-baskit** [reply]
This works for me, wondering why my neovim eslint doesn't show anything (trouble, error, suggestion, or something) and with this now it works

**Reply by nizarhmain** [reply]
Worked for me too, thank you!!!

**Reply by maciej-w** [reply]
Worked for me too. Thanks!

**Reply by nerap** [reply]
Working too well..., interesting

**Comment by gtarsia** [comment]
I spent two hours before finding the solution by @Demianeen. Wondering what's the best solution for this since it's been a month now.

Maybe lazyvim should have a warning in the readme (I can help with this)? Or mention this to Mason devs so they can use 4.8.0 as latest? Not sure what's the best way to spare lazyvim users this problems since there is no error message to figure out.

**Comment by kesor** [comment]
Yesterday `lspconfig` broke it with https://github.com/neovim/nvim-lspconfig/pull/3183

**Comment by esmuellert** [comment]
I tried to attach to VSCode ESLint language server Node process to live debug and figure out the correct config for flat config. I created an ESLint LSP plugin for easier connecting to ESLint server, but the config for flat config should also work for lspconfig.
Docs: https://github.com/esmuellert/nvim-eslint?tab=readme-ov-file#settings-options

Feel free to try out my plugin :)
